src/test/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/RecreateAppealPdfHandlerTest.java

```diff
@@ -38,96 +44,84 @@
-
-        when(emailService.generateUniqueEmailId(caseData.getAppeal().getAppellant())).thenReturn("Test");
+        when(emailService.generateUniqueEmailId(caseDetails.getCaseData().getAppeal().getAppellant())).thenReturn("Test");
 
-        boolean handled = eventService.handleEvent(CREATE_APPEAL_PDF, caseData);
+        PreSubmitCallbackResponse<SscsCaseData> response = recreateAppealPdfHandler.handle(ABOUT_TO_SUBMIT, callback);
 
-        assertTrue(handled);
-        assertEquals("No", caseData.getEvidencePresent());
-        assertNotNull(caseData.getAppeal().getAppellant().getAppointee());
+        assertEquals("No", response.getData().getEvidencePresent());
+        assertNotNull(caseDetails.getCaseData().getAppeal().getAppellant().getAppointee());
 
-        verify(emailService, times(2)).generateUniqueEmailId(eq(caseData.getAppeal().getAppellant()));
-        verify(sscsPdfService,times(1)).generateAndSendPdf(eq(caseData), any(), eq(idamTokens), any());
+        verify(emailService, times(2)).generateUniqueEmailId(eq(caseDetails.getCaseData().getAppeal().getAppellant()));
+        verify(sscsPdfService,times(1)).generateAndSendPdf(eq(caseDetails.getCaseData()), any(), eq(idamTokens), any());
     }
 
     @Test
     public void shouldCallPdfServiceWhenNoAppointee() throws CcdException {
 
-        SscsCaseData caseData = buildCaseDataWithoutPdf();
-
-        when(emailService.generateUniqueEmailId(caseData.getAppeal().getAppellant())).thenReturn("Test");
+        when(emailService.generateUniqueEmailId(caseDetails.getCaseData().getAppeal().getAppellant())).thenReturn("Test");
 
-        caseData.getAppeal().getAppellant().getAppointee().setName(null);
+        caseDetails.getCaseData().getAppeal().getAppellant().getAppointee().setName(null);
 
-        boolean handled = eventService.handleEvent(CREATE_APPEAL_PDF, caseData);
+        PreSubmitCallbackResponse<SscsCaseData> response = recreateAppealPdfHandler.handle(ABOUT_TO_SUBMIT, callback);
 
-        assertTrue(handled);
-        assertEquals("No", caseData.getEvidencePresent());
-        assertNull(caseData.getAppeal().getAppellant().getAppointee());
+        assertEquals("No", caseDetails.getCaseData().getEvidencePresent());
+        assertNull(caseDetails.getCaseData().getAppeal().getAppellant().getAppointee());
 
-        verify(emailService, times(2)).generateUniqueEmailId(eq(caseData.getAppeal().getAppellant()));
-        verify(sscsPdfService).generateAndSendPdf(eq(caseData), any(), eq(idamTokens), any());
+        verify(emailService, times(2)).generateUniqueEmailId(eq(caseDetails.getCaseData().getAppeal().getAppellant()));
+        verify(sscsPdfService).generateAndSendPdf(eq(caseDetails.getCaseData()), any(), eq(idamTokens), any());
     }
 
     @Test
     public void shouldNotCallPdfService() throws CcdException {
 
-        SscsCaseData caseData = buildCaseDataWithPdf();
+        SscsCaseData caseDataWithPdf = buildCaseDataWithPdf();
```

Comment:
Not clear why this test does not call the pdfService.

Reply:
I think it is because the case already has the pdf so no need to generate it again